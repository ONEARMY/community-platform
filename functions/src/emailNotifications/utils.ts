import { CONFIG } from '../config/config'
import { firebaseAuth } from '../Firebase/auth'
import { db } from '../Firebase/firestoreDB'
import { DB_ENDPOINTS } from '../models'
import {
  PK_PROJECT_IMAGE,
  PK_PROJECT_NAME,
  PK_SIGNOFF,
  PP_PROJECT_IMAGE,
  PP_PROJECT_NAME,
  PP_SIGNOFF,
} from './constants'

import type { NotificationType } from 'oa-shared'
import type { IMessageDB, INotification, IUserDB } from '../models'

export const errors = {
  MESSAGE_LIMIT:
    'Emailing of new message blocked: User exceeded email message count',
  NO_ATTACHED_USER:
    'Emailing of new message blocked: Email address not attached to user record',
  PROFILE_NOT_CONTACTABLE:
    'Emailing of new message blocked: Profile not contactable',
  USER_BLOCKED: 'Emailing of new message blocked: User blocked from messaging',
}
const EMAIL_ADDRESS_SEND_LIMIT = 100

export const getUserEmail = async (uid: string): Promise<string | null> => {
  try {
    const { email } = await firebaseAuth.getUser(uid)
    return email
  } catch (error) {
    return null
  }
}

export const getUserAndEmail = async (userName: string) => {
  if (!userName) {
    throw new Error('Cannot get email for empty user name')
  }
  const toUserDoc = (await db
    .collection(DB_ENDPOINTS.users)
    .doc(userName)
    .get()) as FirebaseFirestore.DocumentSnapshot<IUserDB>

  const toUser = toUserDoc.exists ? toUserDoc.data() : undefined
  const toUserEmail = toUser ? await getUserEmail(toUser._authID) : undefined

  if (!toUser || !toUserEmail) {
    throw new Error(`Cannot get user ${userName}`)
  }

  return { toUser, toUserEmail }
}

export const isBelowMessageLimit = async (email) => {
  const { docs } = await db
    .collection(DB_ENDPOINTS.messages)
    .where('email', '==', email)
    .get()
  if (docs.length >= EMAIL_ADDRESS_SEND_LIMIT) {
    throw new Error(errors.MESSAGE_LIMIT)
  }
  return true
}

export const isReceiverContactable = async (userName) => {
  const { toUser } = await getUserAndEmail(userName)
  if (
    typeof toUser.isContactableByPublic === 'boolean' &&
    !toUser.isContactableByPublic
  ) {
    throw new Error(errors.PROFILE_NOT_CONTACTABLE)
  }
  return true
}

export const isSameEmail = (userDoc, email) => {
  if (userDoc.email !== email) {
    throw new Error(errors.NO_ATTACHED_USER)
  }
  return true
}

export const isUserAllowedToMessage = async (uid) => {
  const { docs } = await db
    .collection(DB_ENDPOINTS.users)
    .where('_authID', '==', uid)
    .limit(1)
    .get()

  if (!docs[0]) {
    throw new Error(`Cannot get user ${uid}`)
  }

  if (docs[0].data().isBlockedFromMessaging) {
    throw new Error(errors.USER_BLOCKED)
  }
  return true
}

export const isValidMessageRequest = async ({
  email,
  toUserName,
}: IMessageDB) => {
  const userDoc = await firebaseAuth.getUserByEmail(email)

  try {
    isSameEmail(userDoc, email)
    await isUserAllowedToMessage(userDoc.uid)
    await isBelowMessageLimit(email)
    await isReceiverContactable(toUserName)
    return true
  } catch (error) {
    throw error
  }
}

export const SITE_URL = CONFIG.deployment.site_url

export const getProjectImageSrc = () => {
  switch (SITE_URL) {
    case 'https://dev.onearmy.world':
    case 'https://community.preciousplastic.com':
      return PP_PROJECT_IMAGE
    case 'https://dev.community.projectkamp.com':
    case 'https://community.projectkamp.com':
      return PK_PROJECT_IMAGE
  }
}

export const getProjectName = () => {
  switch (SITE_URL) {
    case 'http://localhost:4000':
    case 'https://dev.onearmy.world':
    case 'https://community.preciousplastic.com':
      return PP_PROJECT_NAME
    case 'https://dev.community.projectkamp.com':
    case 'https://community.projectkamp.com':
      return PK_PROJECT_NAME
  }
}

export const getGreeting = (
  userDisplayName: string,
  greetingMessage: string = '',
) => `
<div align="left" class="greeting-container">
<p>Hey ${userDisplayName}</p>
<p>${greetingMessage}</p>
</div>`

export const getProjectSignoff = () => {
  switch (SITE_URL) {
    case 'https://dev.onearmy.world':
    case 'https://community.preciousplastic.com':
      return PP_SIGNOFF
    case 'https://dev.community.projectkamp.com':
    case 'https://community.projectkamp.com':
      return PK_SIGNOFF
  }
}

export const getUserLink = (displayName: string, userName: string) =>
  `<a href='${SITE_URL}/u/${userName}'>${displayName}</a>`

const HOWTO_NOTIFICATIONS: NotificationType[] = [
  'new_comment',
  'howto_useful',
  'howto_mention',
  'howto_approved',
  'howto_needs_updates',
]
const RESEARCH_NOTIFICATIONS = [
  'new_comment_research',
  'research_useful',
  'research_mention',
  'research_update',
  'research_approved',
  'research_needs_updates',
]
const MAP_PIN_NOTIFICATIONS = ['map_pin_approved', 'map_pin_needs_updates']

const getResourceLabel = (type: NotificationType) => {
  if (HOWTO_NOTIFICATIONS.includes(type)) {
    return 'how-to'
  }
  if (RESEARCH_NOTIFICATIONS.includes(type)) {
    return 'research'
  }
  if (MAP_PIN_NOTIFICATIONS.includes(type)) {
    return 'map pin'
  }
  return 'item'
}

const getResourceLink = (
  notificationType: NotificationType,
  relevantUrl: string,
) =>
  `<a href='${SITE_URL}${relevantUrl}'>${getResourceLabel(
    notificationType,
  )}</a>`

const getCommentListItem = (notification: INotification) => `
<p>
  New comment by ${getUserLink(
    notification.triggeredBy.displayName,
    notification.triggeredBy.userId,
  )} on <a href='${SITE_URL}${notification.relevantUrl}'>${
  notification.title
}</a>
</p>`

const getMentionListItem = (notification: INotification) => `
<p>
  ${getUserLink(
    notification.triggeredBy.displayName,
    notification.triggeredBy.userId,
  )} mentioned you in this ${getResourceLink(
  notification.type,
  notification.relevantUrl,
)}
</p>`

const getUsefulListItem = (notification: INotification) => `
<p>
    ${getUserLink(
      notification.triggeredBy.displayName,
      notification.triggeredBy.userId,
    )} found your ${getResourceLink(
  notification.type,
  notification.relevantUrl,
)}
useful
</p>
`

const getUpdateListItem = (notification: INotification) => `
<p>
    ${getUserLink(
      notification.triggeredBy.displayName,
      notification.triggeredBy.userId,
    )} posted an update to this ${getResourceLink(
  notification.type,
  notification.relevantUrl,
)}
you follow
</p>
`

const getModerationApprovedListItem = (notification: INotification) => `
<p>
    Your ${getResourceLink(
      notification.type,
      notification.relevantUrl,
    )} has been approved
</p>`

const getModerationRejectedListItem = (notification: INotification) => `
<p>
    Your ${getResourceLink(
      notification.type,
      notification.relevantUrl,
    )} needs updates
</p>`

const isCommentNotification = (notification: INotification) =>
  ['new_comment_research', 'new_comment', 'new_comment_discussion'].includes(
    notification.type,
  )

const isMentionNotification = (notification: INotification) =>
  ['research_mention', 'howto_mention'].includes(notification.type)

const isUsefulNotification = (notification: INotification) =>
  ['research_useful', 'howto_useful'].includes(notification.type)

const isUpdateNotification = (notification: INotification) =>
  notification.type === 'research_update'

const isModerationApprovedNotification = (notification: INotification) =>
  ['howto_approved', 'map_pin_approved', 'research_approved'].includes(
    notification.type,
  )

const isModerationRejectedNotification = (notification: INotification) =>
  [
    'howto_needs_updates',
    'map_pin_needs_updates',
    'research_needs_updates',
  ].includes(notification.type)

export const getNotificationListItem = (
  notification: INotification,
): string => {
  if (isCommentNotification(notification)) {
    return getCommentListItem(notification)
  }
  if (isMentionNotification(notification)) {
    return getMentionListItem(notification)
  }
  if (isUsefulNotification(notification)) {
    return getUsefulListItem(notification)
  }
  if (isUpdateNotification(notification)) {
    return getUpdateListItem(notification)
  }
  if (isModerationApprovedNotification(notification)) {
    return getModerationApprovedListItem(notification)
  }
  if (isModerationRejectedNotification(notification)) {
    return getModerationRejectedListItem(notification)
  }
}
