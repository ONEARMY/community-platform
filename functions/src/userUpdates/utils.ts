import { valuesAreDeepEqual } from '../Utils'

import type { IUserDB } from 'oa-shared/models/user'

export const hasDetailsChanged = (
  prevUser: IUserDB,
  user: IUserDB,
): boolean[] => {
  return [
    prevUser.displayName !== user.displayName,
    hasUserImageChanged(prevUser, user),
    prevUser.badges?.verified !== user.badges?.verified,
    prevUser.badges?.supporter !== user.badges?.supporter,
  ]
}

export const hasLocationDetailsChanged = (
  prevUser: IUserDB,
  user: IUserDB,
): boolean[] => {
  return [
    prevUser.location?.countryCode !== user.location?.countryCode,
    prevUser.country !== user.country,
  ]
}

export const hasDetailsForCommentsChanged = (
  prevUser: IUserDB,
  user: IUserDB,
): boolean => {
  const detailsChanged = [
    ...hasDetailsChanged(prevUser, user),
    ...hasLocationDetailsChanged(prevUser, user),
  ]
  return !!detailsChanged.find((detail) => detail === true)
}

export const hasDetailsForMapPinChanged = (
  prevUser: IUserDB,
  user: IUserDB,
): boolean => {
  const detailsChanged = [
    prevUser._lastActive !== user._lastActive,
    prevUser.about !== user.about,
    prevUser.displayName !== user.displayName,
    prevUser.isContactableByPublic !== user.isContactableByPublic,
    prevUser.profileType !== user.profileType,
    prevUser.workspaceType !== user.workspaceType,
    hasUserTagsChanged(prevUser, user),
    ...hasDetailsChanged(prevUser, user),
    ...hasLocationDetailsChanged(prevUser, user),
  ]
  return !!detailsChanged.find((detail) => detail === true)
}

export const hasUserImageChanged = (
  prevUser: IUserDB,
  user: IUserDB,
): boolean => {
  if (!prevUser.userImage && !user.userImage) return false

  if (prevUser.userImage && user.userImage) {
    return !valuesAreDeepEqual(prevUser.userImage, user.userImage)
  }

  if (prevUser.userImage && !user.userImage) return true
  if (!prevUser.userImage && user.userImage) return true
}

export const hasUserTagsChanged = (
  prevUser: IUserDB,
  user: IUserDB,
): boolean => {
  return !valuesAreDeepEqual(prevUser.tags, user.tags)
}

export const getCreatorImage = (userImage: IUserDB['userImage']) => {
  return userImage?.downloadUrl || null
}

export const getFirstCoverImage = (coverImages: IUserDB['coverImages']) => {
  return coverImages?.[0]?.downloadUrl || null
}
