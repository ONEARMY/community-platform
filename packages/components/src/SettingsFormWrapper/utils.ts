export const routeName = (tabTitle: string) =>
  `/settings/${tabTitle.toLowerCase()}`
