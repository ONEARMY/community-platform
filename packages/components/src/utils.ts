import { faker } from '@faker-js/faker'

import type { IComment } from '.'

export const fakeComment = (commentOverloads: Partial<IComment> = {}) => ({
  _created: faker.date.past().toString(),
  creatorCountry: faker.address.countryCode().toLowerCase(),
  _creatorId: faker.internet.userName(),
  _id: faker.database.mongodbObjectId(),
  creatorName: faker.internet.userName(),
  isUserVerified: faker.datatype.boolean(),
  text: faker.lorem.text(),
  isEditable: faker.datatype.boolean(),
  ...commentOverloads,
})

export const createFakeComments = (
  numberOfComments = 2,
  commentOverloads = {},
): IComment[] =>
  [...Array(numberOfComments).keys()].slice(0).map(() =>
    fakeComment({
      ...commentOverloads,
    }),
  )
