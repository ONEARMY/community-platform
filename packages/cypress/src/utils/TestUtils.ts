export enum Page {
  HOWTO = '/how-to',
  ACADEMY = '/academy',
  EVENTS = '/events',
  SETTINGS = '/settings',
}

export const generatedId = (length: number) => {
  let result = ''
  const characters =
    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
  const charactersLength = characters.length
  for (let i = 0; i < length; i++) {
    result += characters.charAt(Math.floor(Math.random() * charactersLength))
  }
  return result
}

export enum DbCollectionName {
  users = 'users',
  howtos = 'howtos',
}
