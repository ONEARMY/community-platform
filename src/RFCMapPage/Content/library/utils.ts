import { IMapPin, IMapPinGrouped, IFilters, IMapGrouping, IPinProps, IMapPinType } from '../../types'
import { GROUPINGS } from './legacy.maps.groupings'

export const getGroupKey = (pin: IMapPin | IMapGrouping): string => {
  return `${pin.type}${pin.subType ? '-'+pin.subType:''}`
}

interface IMapGroups {
  [name: string]: IMapGrouping
}

const ICONS = {
  'member': 'member',
  'workspace': 'workspace',
  'community-builder': 'community',
  'collection-point': 'collection',
  'machine-builder': 'machine'
}

export const getIconFileName = (pinType: IMapPinType): string => {
  return `map-${ICONS[pinType]}.svg`
}

const getMapGroups = (): IMapGroups => {
  const result: IMapGroups = {}
  GROUPINGS.forEach(p => {
    result[getGroupKey(p)] = {...p, icon: getIconFileName(p.type)}
  })
  return result 
}

const GROUPS = getMapGroups()

export const groupPins = (mapPins: IMapPin[]): IMapPinGrouped => {
  const result: IMapPinGrouped = {}
  mapPins.forEach(p => {
    const id = getGroupKey(p)
    /* istanbul ignore else */
    if (!result[id]) result[id] = []
    result[id].push({...p, icon: getIconFileName(p.type)})
  })
  return result 
}

export const unGroupPins = (pinsGrouped: IMapPinGrouped): IPinProps[] => {
  const result: IPinProps[] = []
  Object.keys(pinsGrouped).forEach(g => result.push(...pinsGrouped[g]))
  return result
}

export const getFilters = (pinsGrouped: IMapPinGrouped): IFilters => {
  const result: IFilters = {}
  Object.keys(pinsGrouped).map(g => result[g] = {
    grouping: GROUPS[g].grouping,
    name: g,
    displayName: GROUPS[g].displayName,
    type: GROUPS[g].type,
    icon: GROUPS[g].icon,
    _count: pinsGrouped[g].length,
    active: true
  })
  return result
}
