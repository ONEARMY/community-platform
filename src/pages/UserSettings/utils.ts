import { impactQuestions } from './content/impactQuestions'

import type { IImpactDataField, IImpactYearFieldList } from 'src/models'
import type { IImpactQuestion } from './content/impactQuestions'

export interface ImpactDataFieldInputs {
  [key: string]: IImpactQuestion
}

interface IInputs {
  [key: string]: IImpactDataField
}

export const transformImpactData = (
  fields: IImpactYearFieldList,
): ImpactDataFieldInputs => {
  const transformed = {} as ImpactDataFieldInputs

  Object.values(fields).forEach((field) => {
    const questionField = impactQuestions.find(
      (question) => question.id === field.id,
    )
    return (transformed[field.id] = {
      ...questionField,
      ...field,
    } as IImpactQuestion)
  })

  return transformed
}

export const transformImpactInputs = (
  inputs: IInputs,
): IImpactYearFieldList => {
  const fields = [] as IImpactYearFieldList

  Object.keys(inputs).forEach((key) => {
    const field = inputs[key]
    const question = impactQuestions.find(({ id }) => id === key)

    if (field && question && field.value) {
      fields.push({
        id: question.id,
        value: field.value,
        isVisible:
          typeof field.isVisible === 'boolean' ? field.isVisible : true,
      })
    }
  })

  return fields
}

export const sortImpactYearDisplayFields = (
  fields: IImpactYearFieldList | undefined,
): IImpactYearFieldList => {
  const sortedFields = [] as IImpactYearFieldList
  if (!fields) {
    return sortedFields
  }

  impactQuestions.forEach((question) => {
    const answer = fields.find((element) => element.id === question.id)
    answer !== undefined ? sortedFields.push(answer) : null
  })
  return sortedFields
}
