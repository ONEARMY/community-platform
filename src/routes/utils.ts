import { tagsService } from 'src/services/tagsService'
import { SUPPORTED_IMAGE_EXTENSIONS } from 'src/utils/storage'

import type { SupabaseClient } from '@supabase/supabase-js'

type Slug = string
type Id = number
type Client = SupabaseClient
type Table = 'questions' | 'news'

export const getMetaFields = async (
  client: Client,
  id: Id,
  tagIds: number[],
) => {
  return await Promise.all([
    client
      .from('useful_votes')
      .select('*', { count: 'exact' })
      .eq('content_id', id)
      .eq('content_type', 'news'),
    client
      .from('subscribers')
      .select('user_id', { count: 'exact' })
      .eq('content_id', id)
      .eq('content_type', 'news'),
    await tagsService.getTags(client, tagIds),
  ])
}

export const incrementViewCount = async (
  client: Client,
  table: Table,
  totalViews: number | undefined,
  id: Id,
) => {
  return await client
    .from(table)
    .update({ total_views: (totalViews || 0) + 1 })
    .eq('id', id)
}

export async function isDuplicateExistingSlug(
  slug: Slug,
  id: Id,
  client: Client,
  table: Table,
) {
  const { data } = await client
    .from(table)
    .select('id,slug')
    .eq('slug', slug)
    .single()

  return !!data?.id && data.id !== id
}

export async function isDuplicateNewSlug(
  slug: Slug,
  client: Client,
  table: Table,
) {
  const { data } = await client
    .from(table)
    .select('slug')
    .eq('slug', slug)
    .single()

  return !!data
}

export async function uploadImage(
  id: Id,
  uploadedImage: File,
  client: SupabaseClient,
  table: Table,
) {
  if (!uploadedImage) {
    return null
  }
  const response = await client.storage
    .from(process.env.TENANT_ID as string)
    .upload(`${table}/${id}/${uploadedImage.name}`, uploadedImage)

  const image = response.data
  const error =
    image === null
      ? new Error(`Uploading image: ${response.error.message}`)
      : null

  return { image, error }
}

export function validateImage(image: File | null) {
  const error =
    image?.type && !SUPPORTED_IMAGE_EXTENSIONS.includes(image.type)
      ? new Error(`Unsupported image extension: ${image.type}`)
      : null
  const valid: boolean = !error

  return { valid, error }
}

export function validateImages(images: File[]) {
  const errors: Error[] = []
  for (const image of images) {
    const { error } = validateImage(image)
    if (error) {
      errors.push(error)
    }
    continue
  }

  return { valid: errors.length === 0, errors }
}
