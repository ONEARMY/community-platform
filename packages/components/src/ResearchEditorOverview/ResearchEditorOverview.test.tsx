import '@testing-library/jest-dom/vitest'

import { describe, expect, it } from 'vitest'

import { render } from '../test/utils'
import {
  Default,
  DraftItem,
  ShowBackToResearchButton,
  ShowCreateUpdateButton,
} from './ResearchEditorOverview.stories'

import type { ResearchEditorOverviewProps } from './ResearchEditorOverview'

describe('ResearchEditorOverview', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Default {...(Default.args as ResearchEditorOverviewProps)} />,
    )

    expect(container).toMatchSnapshot()
  })

  it('links back to research', () => {
    const { getByText } = render(
      <ShowBackToResearchButton
        {...(ShowBackToResearchButton.args as ResearchEditorOverviewProps)}
      />,
    )

    expect(getByText('Back to research')).toBeInTheDocument()
  })

  it('links to create update', () => {
    const { getByText } = render(
      <ShowCreateUpdateButton
        {...(ShowCreateUpdateButton.args as ResearchEditorOverviewProps)}
      />,
    )

    expect(getByText('Create update')).toBeInTheDocument()
  })

  it('handles empty updates', () => {
    const { container } = render(
      <Default
        {...(Default.args as ResearchEditorOverviewProps)}
        updates={[]}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('handles falsey updates', () => {
    const { container } = render(
      <Default
        {...(Default.args as ResearchEditorOverviewProps)}
        updates={null as any}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('handles malformed update item', () => {
    const { getByText } = render(
      <Default
        {...(Default.args as ResearchEditorOverviewProps)}
        updates={
          [
            null,
            {
              isActive: true,
              title: 'Update title',
              slug: 'a-slug',
            },
          ] as any
        }
      />,
    )

    expect(getByText('Update title')).toBeInTheDocument()
  })

  it('displays a Draft label', () => {
    const { getByText } = render(
      <DraftItem {...(DraftItem.args as ResearchEditorOverviewProps)} />,
    )

    expect(getByText('Draft')).toBeInTheDocument()
  })
})
