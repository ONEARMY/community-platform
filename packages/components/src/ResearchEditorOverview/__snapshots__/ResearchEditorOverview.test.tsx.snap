// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResearchEditorOverview > handles empty updates 1`] = `
<div>
  <div
    class="css-sbv6lq-Card"
  >
    <h2
      class="css-1vot7yn-Box"
    >
      Research overview
    </h2>
    <button
      class="css-1wcfy94-Button"
      data-cy="create-update"
      type="button"
    >
      <span
        class="css-a6mlbr-Text"
      >
        <a
          class="css-e0v029-Link"
          href="/research/abc/new-update"
        >
          Create update
        </a>
      </span>
    </button>
    <button
      class="css-l7psd3-Button"
      type="button"
    >
      <span
        class="css-a6mlbr-Text"
      >
        <a
          class="css-e0v029-Link"
          href="/research/abc/edit"
        >
          Back to research
        </a>
      </span>
    </button>
  </div>
</div>
`;

exports[`ResearchEditorOverview > handles falsey updates 1`] = `
<div>
  <div
    class="css-sbv6lq-Card"
  >
    <h2
      class="css-1vot7yn-Box"
    >
      Research overview
    </h2>
    <button
      class="css-1wcfy94-Button"
      data-cy="create-update"
      type="button"
    >
      <span
        class="css-a6mlbr-Text"
      >
        <a
          class="css-e0v029-Link"
          href="/research/abc/new-update"
        >
          Create update
        </a>
      </span>
    </button>
    <button
      class="css-l7psd3-Button"
      type="button"
    >
      <span
        class="css-a6mlbr-Text"
      >
        <a
          class="css-e0v029-Link"
          href="/research/abc/edit"
        >
          Back to research
        </a>
      </span>
    </button>
  </div>
</div>
`;

exports[`ResearchEditorOverview > renders correctly 1`] = `
<div>
  <div
    class="css-sbv6lq-Card"
  >
    <h2
      class="css-1vot7yn-Box"
    >
      Research overview
    </h2>
    <ul
      class="css-vebie1-Box"
    >
      <li
        class="css-1dhyciv-Box"
      >
        <span
          class="css-g89m8z-Box"
        >
          <strong>
            Update 1
          </strong>
        </span>
      </li>
      <li
        class="css-1dhyciv-Box"
      >
        <span
          class="css-g89m8z-Box"
        >
          Update 2
          <a
            class="css-1pw6t78-Link"
            href="/research/abc/edit-update/update-2"
          >
            Edit
          </a>
        </span>
      </li>
      <li
        class="css-1dhyciv-Box"
      >
        <span
          class="css-g89m8z-Box"
        >
          Update 3
          <a
            class="css-1pw6t78-Link"
            href="/research/abc/edit-update/update-3"
          >
            Edit
          </a>
        </span>
      </li>
    </ul>
    <button
      class="css-1wcfy94-Button"
      data-cy="create-update"
      type="button"
    >
      <span
        class="css-a6mlbr-Text"
      >
        <a
          class="css-e0v029-Link"
          href="/research/abc/new-update"
        >
          Create update
        </a>
      </span>
    </button>
    <button
      class="css-l7psd3-Button"
      type="button"
    >
      <span
        class="css-a6mlbr-Text"
      >
        <a
          class="css-e0v029-Link"
          href="/research/abc/edit"
        >
          Back to research
        </a>
      </span>
    </button>
  </div>
</div>
`;
